apps/mobile/src/components/browser/browser-sheet/search-bar/search-bar.tsx

```diff
@@ -0,0 +1,153 @@
+import { useRef } from 'react';
+import { TextInput as RNTextInput } from 'react-native';
+import { KeyboardController } from 'react-native-keyboard-controller';
+import Animated from 'react-native-reanimated';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+import { WebViewNavigation } from 'react-native-webview';
+
+import { t } from '@lingui/macro';
+import { useTheme } from '@shopify/restyle';
+import * as Clipboard from 'expo-clipboard';
+import * as Sharing from 'expo-sharing';
+
+import { Box, TextInput, Theme } from '@leather.io/ui/native';
+
+import { BrowserType } from '../utils';
+import { BrowserNavigationBar } from './browser-navigation-bar';
+import { SearchBarToolbar } from './search-bar-toolbar';
+import { useSearchBarAnimatedStyles } from './use-search-bar-animated-styles';
+
+interface SearchBarProps {
+  textUrl: string;
+  setTextUrl(url: string): void;
+  searchUrl: string;
+  onSubmit(): void;
+  goBack(): void;
+  goForward(): void;
+  navState: WebViewNavigation | null;
+  browserType: BrowserType;
+  resetBrowser(): void;
+  refresh(): void;
+}
+
+const AnimatedBox = Animated.createAnimatedComponent(Box);
+
+export function SearchBar({
+  textUrl,
+  setTextUrl,
+  searchUrl,
+  browserType,
+  navState,
+  onSubmit,
+  goBack,
+  goForward,
+  resetBrowser,
+  refresh,
+}: SearchBarProps) {
+  const theme = useTheme<Theme>();
+  const { bottom } = useSafeAreaInsets();
+  const textInputRef = useRef<RNTextInput>(null);
+
+  const { keyboardAvoidingStyle, searchBarStyle, isUrlFocused, browserNavigationBarStyle } =
+    useSearchBarAnimatedStyles();
+
+  const baseSearchBar = (
```

Comment:
Maybe `baseSearchBar` can be a component in it's own file? Or else use style objects instead of props? 

It seems to use a mixture of a style object and then also lots of props anyway. Some of them seem to be repeated?

```
   style={[
...
        {
          paddingHorizontal: theme.spacing['5'],
          paddingTop: theme.spacing['4'],
          paddingBottom: theme.spacing['4'] + bottom,
        },
      ]}
... 
  pt="4"
      pb="4"
      px="5"
```